scripts/pythondistdeps.py

```diff
@@ -303,11 +322,21 @@ def get_marker_env(dist, extra):
                 warn("Version for {!r} has not been found".format(dist), RuntimeWarning)
                 continue
 
+            # If processing an extras subpackage:
+            #   Check that the extras name is declared in the metadata, or
+            #   that there are some dependencies associated with the extras name
+            if extras_subpackage and extras_subpackage not in dist.extras and not dist.requirements_for_extra(extras_subpackage):
```

Comment:
This comment block and the conditional is different from what we have in Fedora. Where does it come from? Was it adapted to work with importlib.metadat?

Reply:
Yes, it was adapted. It's something I wanted to discuss, let's do a call tomorrow.